component/block_cache/block_cache.go

```diff
@@ -295,6 +299,25 @@ func (bc *BlockCache) Configure(_ bool) error {
 		}
 	}
 
+	if common.GenConfig {
+		log.Info("BlockCache::Configure : config generation started")
```

Comment:
We shall discuss on this model, each component opening the config file and dumping its own config part?
If we can come up with some better approach to do this will be great.

Reply:
Yes maybe we can store the yaml content from each component in a common string and finally in the default-config we can open and dump it. Or some other way maybe we can discuss.